Jenkinsfile

```diff
@@ -49,31 +35,21 @@ node ("default-java || light-java") {
         }
     }
 
+    stage('Analytics') {
+        sh './gradlew check spotbugsmain javadoc'
```

Comment:
Any specific reason we fire one big command instead of sticking with the smaller stages? I think it's a bit easier to see where things fail... 🙄

Reply:
And I still think that publishing artifact for which the tests are failing is a bad idea...

Reply:
I just now saw the last sentence in the description 🙈 Even _if_ the analytics stage is not a safeguard right now it still makes sense to check code and run tests before publishing... That way we just need to make the build fail if a test fails and we're good, aren't we?

Reply:
Whoops, didn't realize there were comments here until after pushing locally, sorry.

So yes, it does make sense to run analytics first. Until they actually matter more it can help speed up testing in the meantime, since the artifacts become available a few minutes earlier (analytics tend to take 5-8 minutes). Aiming to swap later on when things are more stable, when we may also have some gates in (if  < `x%` health then fail the build) if that's OK 👍 

Phase number - that's probably more a personal indulgence than anything, with 5 phases or less they fit into the main UI in the classic view like so:

![image](https://user-images.githubusercontent.com/1063833/86533968-8f801b80-be9a-11ea-8f3f-4461fb742792.png)

With 7 that whole chunk of UI gets moved to below the bottom-most graph on the right side :-( I don't remember if 6 fit 🤔 

That's another lower priority todo - look at tweaking the UI to make things fit better / not go weird as easily (group stages in the classic UI?). I know the stages work out better in Blue Ocean since it has the space, although it gets it from not showing any of the analytics graphs. I'm not even sure _where_ those go in Blue Ocean, since I've played with it so little.

There's also potentially a cost in breaking apart the Gradle phases, but I'm not sure how big it is (starting Gradle multiple times means going through the config phase multiple times, not sure how well the daemon resumes)